### PR TITLE
Add widget to lt_xspec for selecting between multiple input spectra

### PR DESCRIPTION
**Pull request by profxj**
As titled.

Also adds functionality to dynaimcally set gui size
based on number of pixels in the monitor.

No new docs or tests.

### CODE REVIEW

**Comment by ntejos** [comment]
May I review this?

**Comment by profxj** [comment]
Please do!

File: linetools/guis/spec_widgets.py [hunk 1]
```diff
@@ -467,6 +471,23 @@ def on_key(self, event):
...
+                            #QtCore.pyqtRemoveInputHook()
+                            #pdb.set_trace()
+                            #QtCore.pyqtRestoreInputHook()
+                            dline = AbsLine(wrest, linelist=llist, z=z.value)
+                            tspec = XSpectrum1D.from_tuple((self.spec.wavelength, self.spec.flux, self.spec.sig))
+                            tspec.normalize(lconti)
+                            dline.analy['spec'] = tspec
+                            dline.limits.set(iwv)
+                            dline.measure_ew()
+                            mssg = 'Using dummy '+ dline.__repr__()
```
Comment (ntejos) [inline linetools/guis/spec_widgets.py]: change message to `"Assuming line is "` (?)

Comment (ntejos) [inline linetools/guis/spec_widgets.py]: These two lines seem duplicates from 471 and 472. May be best to take them out of the `if` and `else` statements.

**Comment by ntejos** [comment]
I tested the GUI and everything seems good. I'm not so convinced whether we want to start labeling the spectra from `0` to `n-1`, wouldn't make more sense to start from `1` to `n` instead? 

I also left comments on the code while ago (see above)... not sure if you considered them.

File: linetools/guis/spec_widgets.py [hunk 2]
```diff
@@ -1294,3 +1315,66 @@ def reload(self):
...
+        self.scale = screen_scale
+        self.parent = parent  # Required
+        spec = self.parent.orig_spec
+
+        # Create the list
+        if spec_labels is None:
+            if hasattr(spec, 'labels'):
+                spec_labels = spec.labels
+            else:
+                spec_labels = ['{:d}'.format(ii) for ii in range(spec.nspec)]
```
Comment (ntejos) [inline linetools/guis/spec_widgets.py]: `{:d}'.format(ii)` -> `{:d}'.format(ii+1)` (?)

File: linetools/guis/spec_widgets.py (hunk 1 above)
Comment (profxj) [inline linetools/guis/spec_widgets.py]: This is so a user can measure the EW without setting a line.
Have made a small edit.

File: linetools/guis/spec_widgets.py (hunk 2 above)
Comment (profxj) [inline linetools/guis/spec_widgets.py]: I think I prefer Python indexing.  That is the way
the spectra are truly indexed anyhow.

**Comment by profxj** [comment]
Merging

**Pull request merged**